lib/resolvme/vault/client.rb

```diff
@@ -95,6 +96,11 @@ def versioned_path?(path)
       def mount_point(path)
         path.split("/").first + "/"
```

Comment:
trimming the path is fine however the bug only exists when determine the mount point of the given path using `def mount_point(path)`. Adding the trim method and then using it in `read_secret_field` adds noise I think.

It will be suficient to update `def mount_point()` to handle leading `/`.
The presence of a leading slash might also cause a bug in `def update_path_if_kv2()` which also uses `path.split()`